src/app/pages/home/components/cinema-details/cinema-details.component.ts

```diff
@@ -31,6 +43,12 @@ export class CinemaDetailsComponent implements OnInit, OnDestroy {
         this.cdr.markForCheck();
       },
     );
+
+    this.sessionSub = this.sessionService.getSessionById(this.cinema.id, 'cinema').subscribe((data) => {
+      this.sessions = data;
+    });
+
+    this.groupedSessionsByDate = groupCinemaSessionsByDate(this.sessions);
```

Comment:
Probably there will be a mistake. You're trying to group sessions before they are loaded. This part should inside subscribe.
`this.groupedSessionsByDate = groupCinemaSessionsByDate(this.sessions);`